### PR TITLE
Add ctrl + c to copy from grid

### LINKED ISSUE #18315: CTRL + C not working for cells returned from SELECT queries in Enhanced MSSQL Extension for Visual Studio Code
- MSSQL Extension Version: 1.25.0
- VSCode Version: 1.94.2
- OS Version: 10.0.22621

Steps to Reproduce:

1. Run any SELECT query
2. Try to copy cell result with CTRL + C

### PR DESCRIPTION
**Pull request by cssuh**
Adds keybinding to copy results from the grid

Fixes #18315

Note: still need to test on Windows

### CODE REVIEW

File: src/reactviews/pages/QueryResult/table/plugins/copyKeybind.plugin.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        );
+    }
+
+    public destroy() {
+        this.grid.onKeyDown.unsubscribe();
+    }
+
+    private async handleKeyDown(e: KeyboardEvent): Promise<void> {
+        let handled = false;
+        console.log(e.getModifierState);
```
Comment (aasimkhan30) [inline src/reactviews/pages/QueryResult/table/plugins/copyKeybind.plugin.ts:55]: NIT: remove console statements.

**Review by aasimkhan30 — APPROVED** [review]

**Comment by caohai** [comment]
Did you get a chance to test on Windows with the latest iteration?

**Review by caohai — APPROVED** [review]

**Comment by cssuh** [comment]
> Did you get a chance to test on Windows with the latest iteration?

Yeah, for some reason using CMD + C works on windows, but I'm hoping it's just because I'm remoting into my windows machine from my mac (I noticed CMD + V works for paste too, which shouldn't be the case).  I'll be able to test on my windows machine later today

File: src/reactviews/pages/QueryResult/table/plugins/copyKeybind.plugin.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        >,
+    ) {
+        this.uri = uri;
+        this.resultSetSummary = resultSetSummary;
+        this.webViewState = webViewState;
+    }
+
+    public init(grid: Slick.Grid<T>) {
+        this.grid = grid;
+        // this.grid.onKeyDown.subscribe(this.handleKeyDown.bind(this));
```
Comment (Benjin) [inline src/reactviews/pages/QueryResult/table/plugins/copyKeybind.plugin.ts:43]: uncomment

File: src/reactviews/pages/QueryResult/table/plugins/copyKeybind.plugin.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    private async handleKeyDown(e: KeyboardEvent): Promise<void> {
+        let handled = false;
+        let platform = await this.webViewState.extensionRpc.call("getPlatform");
+        if (platform === "darwin") {
+            // Cmd + C
+            if (e.metaKey && e.keyCode === 67) {
+                handled = true;
+                await this.handleCopySelection(
```
Comment (Benjin) [inline src/reactviews/pages/QueryResult/table/plugins/copyKeybind.plugin.ts:60]: does `handleCopySelection()` have a telemetry event?

**Review by Benjin — APPROVED** [review]

File: src/reactviews/pages/QueryResult/table/plugins/copyKeybind.plugin.ts (hunk 3 above)
Comment (cssuh) [reply src/reactviews/pages/QueryResult/table/plugins/copyKeybind.plugin.ts:60]: Yes here: https://github.com/microsoft/vscode-mssql/blob/28fc2b84aecf67cd02860b0131a2ac063d889aa3/src/queryResult/queryResultWebViewController.ts#L139

**Comment by Benjin** [comment]
> Yeah, for some reason using CMD + C works on windows, but I'm hoping it's just because I'm remoting into my windows machine from my mac

Yup, that's probably what's going on.  There's an option in remote desktop on mac to continue using the host (mac) keyboard shortcuts for cut/copy/paste that's likely enabled.

**Comment by cssuh** [comment]
> > Yeah, for some reason using CMD + C works on windows, but I'm hoping it's just because I'm remoting into my windows machine from my mac
> 
> Yup, that's probably what's going on. There's an option in remote desktop on mac to continue using the host (mac) keyboard shortcuts for cut/copy/paste that's likely enabled.

Yep, just tested and everything works as expected on Windows.

**Pull request merged**